torch/testing/_internal/common_methods_invocations.py

```diff
@@ -297,6 +310,13 @@ def supported_dtypes(self, device_type):
         else:
             return self.dtypes
 
+    def supported_backward_dtypes(self, device_type):
+        if device_type == 'cpu':
+            return self.backward_dtypesIfCPU
+        if device_type == 'cuda':
+            return self.backward_dtypesIfROCM if TEST_WITH_ROCM else self.backward_dtypesIfCUDA
```

Comment:
Would you also fix line 309 above while we're here?

Reply:
didn't even realize --> will try and if it incurs more test failure I will open a separate PR

Reply:
extra 160+ test failure were found after making that move (tested in #56646). will create a separate PR

Reply:
Thanks for trying. I imagine there's a failure with how we're acquiring the ROCm dtypes. I think today, for example, we populate them from dtypes, for example, and I think some OpInfos are too aggressive in specifying their base dtypes. Maybe we should be acquiring the default ROCm dtypes not from the base "dtypes" but from the CUDA dtypes?